### PR TITLE
fix(js)!: add argon2i(mod|int) support

### PR DESCRIPTION
**Pull request by blu3beri**
Signed-off-by: blu3beri <blu3beri@proton.me>

- StoreKeyMethod can now be properly created with `argon2i`

### CODE REVIEW

File: wrappers/javascript/aries-askar-shared/src/store/StoreKeyMethod.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  Interactive = 'int',
+  Moderate = 'mod',
+}
+
+export class StoreKeyMethod {
+  private method: KdfMethod
+  private argon2Level?: Argon2Level
+
+  public constructor(method: KdfMethod, argon2Level?: Argon2Level) {
+    if (method == KdfMethod.Kdf && !argon2Level) {
```
Comment (TimoGlastra) [inline wrappers/javascript/aries-askar-shared/src/store/StoreKeyMethod.ts:19]:
```suggestion
    if (method === KdfMethod.Kdf && !argon2Level) {
```

Comment (TimoGlastra) [reply wrappers/javascript/aries-askar-shared/src/store/StoreKeyMethod.ts:19]: Should we add a check if kdf level is Raw and argon2level is defined? Also should we just add more enum values instead? What's the benefit of having kdf and then mod/int instaed of having kdf_mod and kdf_int?

**Review by TimoGlastra — DISMISSED** [review]

File: wrappers/javascript/aries-askar-shared/src/store/StoreKeyMethod.ts (hunk 1 above)
Comment (blu3beri) [reply wrappers/javascript/aries-askar-shared/src/store/StoreKeyMethod.ts:19]: Mainly did it to be inline with how it is done in askar itself and this allows, although not quite good, more kdfs.

Comment (blu3beri) [reply wrappers/javascript/aries-askar-shared/src/store/StoreKeyMethod.ts:19]: I will change it to `Argon2IMod` and `Argon2IInt` as growing with more kdfs would also not be a breaking change then.

File: wrappers/javascript/aries-askar-shared/src/store/StoreKeyMethod.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  private method: KdfMethod
+
+  public constructor(method: KdfMethod) {
+    this.method = method
+  }
+
+  public toUri() {
+    return this.method.toString()
+  }
+}
```
Comment (TimoGlastra) [inline wrappers/javascript/aries-askar-shared/src/store/StoreKeyMethod.ts:18]: Is this for future proofing? We could just use the enum value directly right?!

**Review by TimoGlastra — APPROVED** [review]

File: wrappers/javascript/aries-askar-shared/src/store/StoreKeyMethod.ts (hunk 2 above)
Comment (blu3beri) [reply wrappers/javascript/aries-askar-shared/src/store/StoreKeyMethod.ts:18]: Yes, it is for future proofing. Argon could have a salt or other kdf methods might need some custom logic.

**Pull request merged**